process: general traceability concept (#343)
eclipse-score/score: docs/process/general_concepts/score_traceability_concept.rst

..
   # *******************************************************************************
   # Copyright (c) 2025 Contributors to the Eclipse Foundation
   #
   # See the NOTICE file(s) distributed with this work for additional
   # information regarding copyright ownership.
   #
   # This program and the accompanying materials are made available under the
   # terms of the Apache License Version 2.0 which is available at
   # https://www.apache.org/licenses/LICENSE-2.0
   #
   # SPDX-License-Identifier: Apache-2.0
   # *******************************************************************************

.. _general_concepts_traceability:

Traceability concept
--------------------

Traceability is the key to ensure consistency between work products.

> Further traceability supports impact analysis or in general analysis of any dependencies
> and determining of coverage for requirements and verifcation as well as tracking of
> implementation status.

Reviewer: ```suggestion
Furthermore, traceability supports impact analysis, dependency analysis, coverage determination for requirements and verification, and tracking of implementation status.
```

> Therefore S-CORE requires at least for safety-relevant work products single requirements

Reviewer: ```suggestion
Therefore S-CORE requires at least for safety-relevant work products single-requirement
```

granularity of traceability.

The following figures gives an overview about the traceability between different work products
on different levels and detail.

High level traceability overview
++++++++++++++++++++++++++++++++

The following figure shows the traceability between the major S-CORE work products on each
requirements level. Starting from top, the platform level, going down to feature, component
to the bottom the unit level. The concept is based on the classical V-Cylce approach.

> .. figure:: _assets/score_traceability_model_wp_overview.svg

Reviewer: Any reason you are not using .drawio.svg as shown here: https://eclipse-score.github.io/score/main/guidance/docs-as-code.html#draw-io

Author: Was not part of discussion we used that already in the POC, why did you not consider that?

Reviewer 2: Does this docs-as-code guideline request to use VS code? Cannot another drawio app be used?

Reviewer: > Does this docs-as-code guideline request to use VS code?
That's not intentional. I'll have a look.

> Cannot another drawio app be used?
Yes, but generally we want to have `.drawio.svg` files. I'm not sure whether those are technically different than `.svg` files, or whether that's purely a naming convention. We use `.drawio.svg` to clearly identify that those files can be edited via draw io.

Reviewer: Update: I've tried editing your .svg files. I can edit them. So it's purely a naming convention to store drawio-svgs as `.drawio.svg`.

on different levels and detail.

High level traceability overview
++++++++++++++++++++++++++++++++

The following figure shows the traceability between the major **S-CORE** work products on each
requirements level. Starting from top, the platform level, going down to feature, component
to the bottom the unit level. The concept is based on the classical V-Cylce approach.

Change request are traced to all affected work products.

.. figure:: _assets/score_traceability_model_wp_overview.svg
  :width: 100%
  :align: center
  :alt: High level traceability overview for **S-CORE** work products

  High level traceability overview for **S-CORE** work products

The next figure sets the focus on the feature level and adds the traceability from the Feature
Requirements to the Feature Architecture, Feature Safety Analysis and the Feature Assumption
of use. For convenience also the traceability to upper and lower lever requirements is shown.

.. figure:: _assets/score_traceability_model_feat_overview.svg
  :width: 100%
  :align: center
  :alt: High level traceability overview for **S-CORE** feature work products

  High level traceability overview for **S-CORE** feature work products

> The next figures sets the focus on the component level and adds the traceability from the

Reviewer 2: please split the text and put the picture after each relevant section
